### ISSUE TITLE
Teaching Archive Overhaul: Pagination, Year Filter UI/UX/Accessibility

### ISSUE THREAD

**Issue by JonEsparaz** [themeetinghouse/web#656]
## Pagination

The network requests are massive and UX isn't great since the page takes a while to load.

I think videos/series should be loaded by year. This can be achieved using the `startDate`/`publishedDate` indices:
```typescript
const variables: GetSeriesBySeriesTypeQueryVariables = {
   ...
   startDate: { beginsWith: '2021' }
};

const variables: GetVideoByVideoTypeQueryVariables = {
   ...
   publishedDate: { beginsWith: '2021' }
};
```

The archive can load N years to start, then load another N on scroll. If the client filters by year, and that year has not been loaded, it can be specifically requested. Storing the data in a hash table (JS object) with the year as the key is probably a good idea.

- [ ] Paginate API requests by year
- [ ] Infinite scroll
- [ ] Logic to handle filtering by year (depending on if that year has been loaded or not)

## Year Filter UI/UX/accessibility
- [ ] UI: Rather than pills, this should probably be a dropdown.
- [ ] UX: Hash routing to each year will allow users to share a link for teaching from a specific year. Additionally, when the user "filters" by year, instead of an expensive rerender, the page can just scroll to the hash link.
- [ ] Accessibility: The pills are not keyboard focusable (dropdown will resolve this).
- [ ] Accessibility: The videos/series themselves are not keyboard focusable (this is an issue everywhere on the website).

**Comment by GeorgeBellTMH** [comment]
We can probably load smaller thumbnails...

**Comment by JonEsparaz** [comment]
@GeorgeBellTMH why does almost every issue say edited by GeorgeBellTMH? Do you have a bot or something? 

Asking because it always remove leading whitespace from my code blocks.

**Comment by GeorgeBellTMH** [comment]
It might be the sync to wrike...which we use internally for project management...

**Comment by GeorgeBellTMH** [comment]
(it is...)

**Comment by lucastbelem** [comment]
Closing this for now as the archive is gone..
